src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -162,6 +162,9 @@ public void RunAndBlock(CancellationToken cancellationToken = default(Cancellati
 
                     newInstance.StartAsync(cancellationToken).GetAwaiter().GetResult();
 
+                    // Apply JobHost metadata
+                    newInstance.ApplyJobHostMetadata();
```

Comment:
Something still seems wrong about this. We don't want ScriptHostManager to be messing with this from the outside. All initialization should be done by the host during its create/initialize logic. Also, once the host is created above, ScriptHostManager.CanInvoke will be returning true, which means http requests can come in. Applying disabled state at a later point here means disabled functions can be invoked for a short period.